### ISSUE TITLE
Improve suggestion for adding build-depends

### ISSUE THREAD

**Issue by varungandhi-src** [haskell/cabal#8262]
**Describe the bug**

When trying to build [facebookincubator/Glean](https://github.com/facebookincubator/Glean), I got an error with a suggestion:

```
    Could not load module ‘Glean.Internal.Types’
    It is a member of the hidden package ‘glean-0.1.0.0’.
    Perhaps you need to add ‘glean’ to the build-depends in your .cabal file.
```

This suggestion is incorrect, because the error came about when trying to build the `gen-schema` target in `glean.cabal`.

**To Reproduce**
You can follow the Glean [build instructions](https://glean.software/docs/building/) but that's probably overkill. It should be possible to repro this by creating a minimal version of `glean.cabal`.

**Expected behavior**
In this context, the correct suggestion would be to add a dependency on `glean:if-internal-hs` since that is the only library target which includes `Glean.Internal.Types`.

**System information**
- Ubuntu 22.04
- cabal 3.6.2.0, ghc 8.10.7

**Comment by ffaf1** [comment]
Alas same problem with `cabal-install` 3.9. [MRE](https://github.com/haskell/cabal/files/9032382/mre.zip).

**Comment by Mikolaj** [comment]
So, adding `glean` doesn't fix the compilation failure, only adding `glean:if-internal-hs` does? It sounds like the hinter is too simplistic and assumes there's only one library in the project (as it was in the old times).

**Comment by varungandhi-src** [comment]
> So, adding glean doesn't fix the compilation failure, only adding glean:if-internal-hs does?

Right, it is recommending adding a dependency from a target inside the package to the package itself. Ideally, I think, it should be recommending a specific target in the package, if the package has multiple targets.
